modules/discriminators.py

```diff
@@ -243,8 +250,10 @@ async def _assess_and_backtrack_gpt_factory(
         generations = await comparison_generator(
             agent, middleman_settings_copy, assess_and_backtrack_prompt
         )
+        if generations.outputs is None:
+            raise ValueError("No generations returned from comparison_generator")
```

Comment:
Is this OK?

Reply:
Seems better than what's currently there which will also just error out, but I think ideally we retry the generation.

Reply:
Or actually I think my assumption is that this never fails and in fact hasn't failed. So I wouldn't want to just silently retry. If it fails then something is up. Seems good to explicitly raise the error.